### ISSUE TITLE
508-defect-3 [COGNITION]: Links SHOULD be written as links OR Buttons SHOULD be styled as buttons

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#17740]
# [508-defect-3](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-3)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

<hr/>

## Point of Contact

**VFS Point of Contact:** Josh Kim

## User Story or Problem Statement

As a user, I expect to be able to open a link in a new tab so that I can navigate in a way that works best for me.

## Details

The "go back and answer questions" link is written as a `button` which is materially dishonest. Users may become frustrated that they are unable to open a new tab. Screen reader users may be confused that the text seems more like a link despite being announced as a button.

## Acceptance Criteria

- [ ] "go back and answer questions" is written with an `a` tag 

## Environment

* Operating System: all
* Browser: any
* Screenreading device: any
* Server destination: staging

## Steps to Recreate

1. Enter `https://staging.va.gov/decision-reviews/higher-level-review/request-higher-level-review-form-20-0996/introduction` in browser
2. Right click on "go back and answer questions"
3. Confirm you cannot open in a new tab (it's a button)

## Solution (if known)

Use `a` instead of `button`.

## WCAG or Vendor Guidance (optional)

* [Adam Silver: But sometimes buttons look like links](https://adamsilver.io/articles/but-sometimes-buttons-look-like-links/)
* [VA Design System guidance on buttons](https://design.va.gov/components/buttons#guidance)

### Before
<img width="824" alt="Screen Shot 2020-12-18 at 1 37 29 PM" src="https://user-images.githubusercontent.com/14154792/102650812-782ba680-4139-11eb-9913-c99eceaf1836.png">

### After
<img width="798" alt="Screen Shot 2020-12-18 at 1 59 14 PM" src="https://user-images.githubusercontent.com/14154792/102650824-7cf05a80-4139-11eb-95e8-ce32e652ecf9.png">

**Comment by Mottie** [comment]
Since interacting with "go back and answer questions" doesn't change the page URL, I thought it would be best to make it a button. All it does is clear a value within the `sessionStorage`. From my understanding, links (`a` tags) should take the user to a new page, and buttons should cause an action. If you think it needs to be an `a` link, then I can change it.

**Comment by joshkimux** [comment]
@Mottie Ohhh, I didn't notice that. Thanks for pointing that out! That's tricky then because the _signifier_ (the text itself) is telling the user that this is a link, yet the actual behavior and semantic affordance matches a button.

In this case, I still think we should still make it an `a`, because we should default to the user's mental model, not the system's. I've observed in the wizard research that some users wanted the freedom to open items in new links which exploring, so if this _appears_ to be a link-- we should match that expectation.

Let me know your thoughts on this! If it's alot of extra work, we can let it slide here and maybe it pass it to the design system and content teams to review 😄

**Comment by Mottie** [comment]
So, because the action is to clear the `sessionStorage`, if I changed it to a link then I could set the `href` back to the introduction page (instead of the usual `#`), but then I would have to prevent it from reloading the page... I mean I guess it's okay if it actually does reload the page - I think some of the other wizards do that - but I didn't think it was necessary. Preventing it from working may also block the user from opening the link in a new tab - I always middle click links to open new tabs, I haven't tried right clicking a link like this that blocks default behavior and see if it opens a new tab.

I'm not against changing the tag, but I'm not completely clear on how it should behave - let it reload the page, or prevent reloading and possibly prevent allowing the link to open a new tab?

**Comment by joshkimux** [comment]
This is all good context. Let me get back to you on this. I'm curious to understand the original design intent a bit more. Also curious to get @1Copenut 's thoughts on this too.

**Comment by Mottie** [comment]
Here is the IA flow documentation - https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/information-architecture/ia-reviews/websites-apply-wizard.md#userpage-flows-

And the design - https://vsateams.invisionapp.com/share/W5U21BDFTQK#/screens/404076958

Neither explicitly says that the wizard and the introduction page content have to be on the same URL, but unlike HLR (since it's brand new), most forms already have links pointing to the `/introduction` url as a starting point.

**Comment by joshkimux** [comment]
Thank you for sharing this! The IA flow documentation states "When the user clicks this option, it should function like the browser 'Back' button - they should be returned to the original /introduction page with the wizard completely reset (i.e. their previous answers are cleared)." In which case, reloading the page would match the original (at least written) intent.

Since I didn't design the wizard originally, maybe we should probably rope in the architect at some point and flag this as something for the design system team to handle?

**Comment by Mottie** [comment]
Fixed in https://github.com/department-of-veterans-affairs/vets-website/pull/15445 - should be available in staging shortly.

**Comment by joshkimux** [comment]
@Mottie You're the MVP! Thank you!

**Comment by joshkimux** [comment]
Just verified this has been fixed. Closing!
